snap/validate.go

```diff
@@ -81,6 +82,99 @@ func ValidateAlias(alias string) error {
 	return nil
 }
 
+// ValidateAppSocketName checks if a string ca be used as a name for a socket
+// (for socket activation).
+func ValidateAppSocketName(name string) error {
+	valid := validSnapName.MatchString(name)
+	if !valid {
+		return fmt.Errorf("invalid app socket name: %q", name)
+	}
+	return nil
+}
+
+// ValidateAppSocketListenAddress checks that the value of socket addresses.
+func ValidateAppSocketListenAddress(socket *SocketInfo, fieldName string, address string) error {
+	if address == "" {
+		return fmt.Errorf("socket %q must define %q", socket.Name, fieldName)
+	}
+
+	switch address[0] {
+	case '/', '$':
+		return validateAppSocketListenAddressPath(socket, fieldName, address)
+	case '@':
+		return validateAppSocketListenAddressAbstractSocket(socket, fieldName, address)
+	default:
+		return validateAppSocketListenAddressNetAddress(socket, fieldName, address)
+	}
+}
+
+func validateAppSocketListenAddressPath(socket *SocketInfo, fieldName string, path string) error {
```

Comment:
We must ensure a clean path before we can go forward here, otherwise people might use .. to get out of the specified locations. Checking that path == filepath.Clean(path) should do the job.

Reply:
done